UnityProject/Assets/Scripts/GunScriptSystems.cs

```diff
@@ -890,15 +891,21 @@ public class CylinderHammerCycleSystem : GunSystemBase {
         public override void Update() {
             if (rcc.hammer_cycling && hc.thumb_on_hammer != Thumb.SLOW_LOWERING) {
                 if (hc.hammer_cocked > 0.0f) {
-                    if (rcc.is_closed && hc.hammer_cocked == 1.0f && hc.prev_hammer_cocked != 1.0f) {
+                    if (rcc.is_closed && hc.hammer_cocked == 1.0f && hc.prev_hammer_cocked != 1.0f && !needsToReset) {
+                        last_pos = (rcc.active_cylinder + hc.hammer_cocked) * 360 / rcc.cylinder_capacity;
+                        needsToReset = true;
                         ++rcc.active_cylinder;
                         rcc.cylinder_rotation = rcc.active_cylinder * 360.0f / rcc.cylinder_capacity;
                     }
-                    if (rcc.is_closed && hc.hammer_cocked < 1.0f) {
+                    if (rcc.is_closed && hc.hammer_cocked < 1.0f && (rcc.active_cylinder + hc.hammer_cocked) * 360.0f / rcc.cylinder_capacity > last_pos && !needsToReset) {
+                        last_pos = (rcc.active_cylinder + hc.hammer_cocked) * 360.0f / rcc.cylinder_capacity;
```

Comment:
`(rcc.active_cylinder + hc.hammer_cocked) * 360.0f / rcc.cylinder_capacity` could be saved to a temporary variable called `current_pos`, to avoid having duplicate code for this calculation.

Reply:
Agreed